src/main/java/seedu/address/logic/CommandHistory.java

```diff
@@ -25,6 +26,22 @@ public CommandHistory(CommandHistory commandHistory) {
     public void add(String userInput) {
         requireNonNull(userInput);
         userInputHistory.add(userInput);
+
+        /**
+         * Converts userInputHistory from a linked list into a string so that it can be passed to the CommandArchive
+         * class where the latest command can be extracted and added to commandFile.txt which tracks all the keystrokes
+         * of a user with a timestamp
+         */
+        String newLine = System.getProperty("line.separator");
+        StringBuilder string = new StringBuilder();
+        Iterator<?> it = userInputHistory.descendingIterator();
```

Comment:
There is no need to use a wildcard generic here. Since you know userInputHistory is a List\<String\>, it suffices to use Iterator\<String\>. As a general rule, avoid using the wildcard generic on its own.  It is as dangerous as using the Object type.

Also, 'it' is not a good variable name.